vnext/JSI/Universal/ChakraRtRuntimeFactory.cpp

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef CHAKRACORE
+
+#include "ChakraRtRuntimeFactory.h"
+
+#include "ChakraRtRuntime.h"
+#include "ChakraRuntimeArgs.h"
+
+namespace Microsoft::JSI {
+
+std::unique_ptr<facebook::jsi::Runtime> makeChakraRtRuntime(const std::shared_ptr<ChakraRtRuntimeArgs> &args) noexcept {
+  auto runtime = std::make_unique<ChakraRtRuntime>(args);
+  runtime->Initialize();
```

Comment:
>Initialize [](start = 11, length = 10)

Why don't we do this in the ctor if we want to do this after building anyway? The main place I've seen two-phase intiaalization be useful is for cases where we want to delay init because something is expensive, or offer flow control to handle failure.

Reply:
This would also let us get rid of the factory, unless we wanted it for a specific reason.

---
In reply to: [345416842](https://github.com/microsoft/react-native-windows/pull/3650#discussion_r345416842) [](ancestors = 345416842)

Reply:
Initialize() need to invoke EvaluateJavaScriptSimple, a virtual function, on a bootstrap bundle for implementing HostObjects. Since virtual dispatch does not work in constructors, we need a separate Initialize step.

---
In reply to: [345417135](https://github.com/microsoft/react-native-windows/pull/3650#discussion_r345417135) [](ancestors = 345417135,345416842)